antenna/app.py

```diff
@@ -3,223 +3,211 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 import cgi
+import hashlib
 import io
 import json
 import logging
 import time
 import zlib
 
 from everett.manager import ConfigManager, ConfigOSEnv, parse_class
+from everett.component import ConfigOptions, RequiredConfigMixin
 import falcon
 
 from antenna.lib.datetimeutil import utc_now
 from antenna.lib.ooid import create_new_ooid
-from antenna.lib.storage import Storage
-from antenna.throttler import DISCARD, IGNORE
+from antenna.util import de_null
+
 
 logger = logging.getLogger('gunicorn.error')
 
 
-# FIXME: Figure out whether we need to implement this or nix it. For now, we're
-# just going to define it as a synonym.
-MemoryDumpsMapping = dict
+class BreakpadSubmitterResource(RequiredConfigMixin):
+    """Handles incoming breakpad crash reports and saves to S3"""
+    required_config = ConfigOptions()
+    required_config.add_option(
+        'dump_field', default='upload_file_minidump',
+        doc='the name of the field in the POST data for dumps'
+    )
+
+    required_config.add_option(
+        'dump_id_prefix', default='bp-',
+        doc='the crash type prefix'
+    )
 
+    required_config.add_option(
+        'crashstorage_class',
+        default='antenna.external.boto.crashstorage.BotoS3CrashStorage',
+        parser=parse_class,
+        doc='the class in charge of storing crashes'
+    )
```

Comment:
I'm implementing all the crashstorage stuff after this lands. This config and the usage of it is just filler for now.